src/Compiler/Facilities/BuildGraph.fs

```diff
@@ -151,24 +151,7 @@ type NodeCode private () =
         NodeCode.RunImmediate(computation, CancellationToken.None)
 
     static member StartAsTask_ForTesting(computation: NodeCode<'T>, ?ct: CancellationToken) =
-        let diagnosticsLogger = DiagnosticsThreadStatics.DiagnosticsLogger
-        let phase = DiagnosticsThreadStatics.BuildPhase
-        let ct2 = Cancellable.Token
-
-        try
-            let work =
-                async {
-                    DiagnosticsThreadStatics.DiagnosticsLogger <- diagnosticsLogger
-                    DiagnosticsThreadStatics.BuildPhase <- phase
-                    Cancellable.Token <- ct2
-                    return! computation |> Async.AwaitNodeCode
-                }
-
-            Async.StartAsTask(work, cancellationToken = defaultArg ct CancellationToken.None)
-        finally
-            DiagnosticsThreadStatics.DiagnosticsLogger <- diagnosticsLogger
-            DiagnosticsThreadStatics.BuildPhase <- phase
-            Cancellable.Token <- ct2
+        Async.StartAsTask(computation |> Async.AwaitNodeCode, ?cancellationToken = ct)
```

Comment:
Removing these bits looks suspicious to me. It's what keeps the needed parts of the compiler context when it switches threads.

Reply:
I think I just replaced it with what we use in production? (AwaitNodeCode)

Reply:
Now that I look at it, this code is different than `wrapThreadStaticInfo`.
Here `DiagnosticsThreadStatics` of the calling thread are passed onto the new thread.
In `wrapThreadStaticInfo` they are not:
https://github.com/dotnet/fsharp/blob/3af3d4182ea5284f37cc85f408f392070eb1ee8c/src/Compiler/Facilities/BuildGraph.fs#L16-L26